samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -515,54 +515,28 @@ private AssessmentGradingData persistAssessmentGrading(ActionEvent ae,
 			ItemGradingData newItem = iter1.next();
 			ItemGradingData oldItem = map.get(newItem
 					.getItemGradingId());
-			if (oldItem != null) {
-				// itemGrading exists and value has been change, then need
-				// update
-				Boolean oldReview = oldItem.getReview();
-				Boolean newReview = newItem.getReview();
-				Long oldAnswerId = oldItem.getPublishedAnswerId();
-				Long newAnswerId = newItem.getPublishedAnswerId();
-				String oldRationale = oldItem.getRationale();
-				String newRationale = TextFormat.convertPlaintextToFormattedTextNoHighUnicode(log, newItem.getRationale());
-				String oldAnswerText = oldItem.getAnswerText();
-				// Change to allow student submissions in rich-text [SAK-17021]
-				String newAnswerText = ContextUtil.stringWYSIWYG(newItem.getAnswerText());
-				if ((oldReview != null && !oldReview.equals(newReview))
-				    || (newReview!=null && !newReview.equals(oldReview))
-						|| (oldAnswerId != null && !oldAnswerId
-								.equals(newAnswerId))
-						|| (newAnswerId != null && !newAnswerId
-								.equals(oldAnswerId))
-						|| (oldRationale != null && !oldRationale
-								.equals(newRationale))
-						|| (newRationale != null && !newRationale
-								.equals(oldRationale))
-						|| (oldAnswerText != null && !oldAnswerText
-								.equals(newAnswerText))
-						|| (newAnswerText != null && !newAnswerText
-								.equals(oldAnswerText))
-						|| fibMap.get(oldItem.getPublishedItemId()) != null
-						|| emiMap.get(oldItem.getPublishedItemId()) != null 
-						|| finMap.get(oldItem.getPublishedItemId())!=null
-						|| calcQuestionMap.get(oldItem.getPublishedItemId())!=null
-						|| mcmrMap.get(oldItem.getPublishedItemId()) != null) {
-					oldItem.setReview(newItem.getReview());
-					oldItem.setPublishedAnswerId(newItem.getPublishedAnswerId());
-					oldItem.setRationale(newRationale);
-							
-					oldItem.setAnswerText(newAnswerText);
-					oldItem.setSubmittedDate(new Date());
-					oldItem.setAutoScore(newItem.getAutoScore());
-					oldItem.setOverrideScore(newItem.getOverrideScore());
-					updateItemGradingSet.add(oldItem);
-					// log.debug("**** SubmitToGrading: need update
-					// "+oldItem.getItemGradingId());
-				}
+			if (oldItem != null && !oldItem.equals(newItem) ||
```

Comment:
I believe this is where the real fix for this issue is. I thought it was in the equals method of ItemGradingData but it looks like it just was because it wasn't using that method, rather rechecking it all here.